osu.Game.Rulesets.Soyokaze/Mods/SoyokazeModHidden.cs

```diff
@@ -65,9 +80,12 @@ private void applyState(DrawableHitObject drawableObject, bool increaseVisibilit
                 case DrawableHitCircle circle:
                     fadeTarget = circle.HitCircle;
                     if (!increaseVisibility)
-                        using (circle.BeginAbsoluteSequence(hitObject.StartTime - hitObject.Preempt))
+                        using (circle.BeginAbsoluteSequence(FadingApproachCircle.Value? fadeOutStartTime : hitObject.StartTime - hitObject.Preempt))
                         {
-                            circle.ApproachCircle.Hide();
+                            if (FadingApproachCircle.Value)
+                                circle.ApproachCircle.FadeOut(fadeOutDuration / FadingApproachCircleSpeed.Value);
+                            else
+                                circle.ApproachCircle.Hide();
                         }
```

Comment:
The `if` logic seems a bit clunky, you're checking `FadingApproachCircle.Value` twice (once in the ternary statement, and once again below). This should be separated into two separate `if`/`else` blocks, each with its own `using` block.